### PR TITLE
map-based configuration

### PR DESCRIPTION
**Pull request by lcrilly**

### CODE REVIEW

File: openid_connect.js [hunk 1]
```diff
@@ -1,90 +1,42 @@
...
+
+        // Choose a nonce for this flow for the client, and hash it for the IdP
+        var noncePlain = r.variables.request_id;
+        var c = require('crypto');
+        var h = c.createHmac('sha256', r.variables.oidc_hmac_key).update(noncePlain);
+        var nonceHash = h.digest('base64url');
 
-function oidcRefreshRequest(r) {
+        // Redirect the client to the IdP login page with the headers we need for state
+        r.headersOut['Set-Cookie'] = "auth_redir=" + r.variables.request_uri + "; " + r.variables.oidc_cookie_flags;
```
Comment (xeioex) [inline openid_connect.js]: // Style suggestion

`r.headersOut['Set-Cookie'] = "auth_redir=" + r.variables.request_uri + "; " + r.variables.oidc_cookie_flags;`

->

`` r.headersOut['Set-Cookie'] = `auth_redir=${r.variables.request_uri}; ${r.variables.oidc_cookie_flags}`; ``

``
   r.return(302, `${r.variables.oidc_authz_endpoint}?response_type=code&scope=${r.variables.oidc_scopes }&client_id=${r.variables.oidc_client}&state=0&redirect_uri=${r.variables.scheme}://${r.variables.host}:${r.variables.server_port}${r.variables.redir_location}&nonce=${nonceHash});
``

File: openid_connect.js [hunk 2]
```diff
@@ -144,6 +96,7 @@ function oidcRefreshRequest(r) {
                             r.variables.refresh_token = tokenset.refresh_token; // Update key-value store
                         }
 
+                        r.headersOut["WWW-Authenticate"] = ""; // Remove evidence of original failed auth_jwt
```
Comment (xeioex) [inline openid_connect.js]: // Style

`delete r.headersOut["WWW-Authenticate"]` also works.

File: openid_connect.js [hunk 3]
```diff
@@ -1,90 +1,42 @@
...
-    // proxied to the IdP in exchange for a JWT
-    r.subrequest("/_token", "code=" + r.variables.arg_code,
-        function(reply) {
-            if (reply.status == 504) {
-                r.error("OIDC timeout connecting to IdP when sending authorization code");
-                r.return(504);
-                return;
-            }
+function oidcAuth(r) {
+    if (r.variables.refresh_token == null || r.variables.refresh_token == "-") {
```
Comment (xeioex) [inline openid_connect.js]: // Style

`r.variables.refresh_token == null` -> `!r.variables.refresh_token` // A bit shorter

File: openid_connect_configuration.conf [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# OpenID Connect configuration
```
Comment (thresheek) [inline openid_connect_configuration.conf]: The file itself is not included from frontend.conf nor openid_connect.server_conf.

File: openid_connect.js [hunk 5]
```diff
@@ -156,10 +109,83 @@ function oidcRefreshRequest(r) {
...
+                            r.variables.new_refresh = tokenset.refresh_token; // Create key-value store entry
+                            r.log("OIDC refresh token stored");
+                        } else {
+                            r.warn("OIDC no refresh token");
+                        }
+
+                        // Add opaque token to keyval session store
+                        r.log("OIDC success, creating session " + r.variables.request_id);
+                        r.variables.new_session = tokenset.id_token; // Create key-value store entry
+                        r.headersOut["Set-Cookie"] = "auth_token=" + r.variables.request_id + "; Path=/; HttpOnly;";
```
Comment (lcrilly) [inline openid_connect.js]: This does not use the `oidc_cookie_flags` variable as intended.

**Pull request closed without merging**